daily_snapshot_terraform/modules/daily_snapshot/service/init.sh

```diff
@@ -24,3 +24,24 @@ chmod 777 forest_db
 
 # Setup cron jobs
 cp calibnet_cron_job mainnet_cron_job /etc/cron.hourly/
+
+# Set-up the New Relic license key and custom configuration
+cat << EOF | sudo tee -a /etc/newrelic-infra.yml
+enable_process_metrics: true
+status_server_enabled: true
+status_server_port: 18003
+license_key: "$NR_LICENSE_KEY"
+custom_attributes:
+  nr_deployed_by: newrelic-cli
+display_name: "$NAME"
+EOF
+
+# Add the New Relic infrastructure monitoring agent repository to the system's list of yum repos.
+# The resulting file (/etc/yum.repos.d/newrelic-infra.repo) is used by yum to locate the New Relic packages.
+sudo curl -o /etc/yum.repos.d/newrelic-infra.repo https://download.newrelic.com/infrastructure_agent/linux/yum/el/9/x86_64/newrelic-infra.repo
+
+# Refreshes the New Relic repository. This step ensures that yum is aware of the latest versions of available packages.
+sudo yum -q makecache -y --disablerepo='*' --enablerepo='newrelic-infra'
+
+# Installs the New Relic infrastructure agent. This package provides the monitoring functionality needed.
+sudo yum install newrelic-infra -y
```

Comment:
`yum` should not be used anymore in Fedora. If there's a compelling reason to use `yum` instead of `dnf`, it must be documented.